pkg/apis/v1beta1/nodepool_validation.go

```diff
@@ -94,18 +98,52 @@ func (in *NodeClaimTemplate) validateRequirementsNodePoolKeyDoesNotExist() (errs
 	return errs
 }
 
+//nolint:gocyclo
 func (in *Disruption) validate() (errs *apis.FieldError) {
 	if in.ExpireAfter.Duration != nil && *in.ExpireAfter.Duration < 0 {
-		return errs.Also(apis.ErrInvalidValue("cannot be negative", "expirationTTL"))
+		return errs.Also(apis.ErrInvalidValue(in.ExpireAfter.Duration, "expirationTTL", "cannot be negative"))
 	}
 	if in.ConsolidateAfter != nil && in.ConsolidateAfter.Duration != nil && *in.ConsolidateAfter.Duration < 0 {
-		return errs.Also(apis.ErrInvalidValue("cannot be negative", "consolidationTTL"))
+		return errs.Also(apis.ErrInvalidValue(in.ConsolidateAfter.Duration, "consolidationTTL", "cannot be negative"))
 	}
 	if in.ConsolidateAfter != nil && in.ConsolidateAfter.Duration != nil && in.ConsolidationPolicy == ConsolidationPolicyWhenUnderutilized {
 		return errs.Also(apis.ErrGeneric("consolidateAfter cannot be combined with consolidationPolicy=WhenUnderutilized"))
 	}
 	if in.ConsolidateAfter == nil && in.ConsolidationPolicy == ConsolidationPolicyWhenEmpty {
 		return errs.Also(apis.ErrGeneric("consolidateAfter must be specified with consolidationPolicy=WhenEmpty"))
 	}
+	if len(in.Budgets) > 50 {
+		return errs.Also(apis.ErrGeneric("budgets must have a length of at most 50"))
+	}
+	for i := range in.Budgets {
+		budget := in.Budgets[i]
+		if err := budget.validate(); err != nil {
+			errs = errs.Also(err.ViaIndex(i).ViaField("budget"))
+		}
+	}
+	return errs
+}
+
+//nolint:gocyclo
+func (in *Budget) validate() (errs *apis.FieldError) {
+	if (in.Crontab != nil && in.Duration == nil) || (in.Crontab == nil && in.Duration != nil) {
+		return apis.ErrGeneric("crontab and duration must be specified together")
+	}
+	if matched, err := regexp.MatchString("^((100|[0-9]{1,2})%|[0-9]+)$", in.MaxUnavailable); err != nil || !matched {
+		return apis.ErrInvalidValue(in.MaxUnavailable, "maxUnavailable", "must be a valid intorstring")
+	}
+	if in.Crontab != nil {
+		if _, err := cron.ParseStandard(lo.FromPtr(in.Crontab)); err != nil {
+			return apis.ErrInvalidValue(in.Crontab, "crontab", fmt.Sprintf("invalid crontab %s", err))
+		}
+	}
+	if in.Duration != nil {
```

Comment:
Can we double-check if we actually need to check some of these things, given that things that are just OpenAPI validations and aren't part of CEL can just be passed off to the apiserver. I'd like to avoid duplicating checks here twice if the request wouldn't even succeed at the apiserver

Reply:
Discussed offline. Checking that durations are under a minute should be enforced at the pattern matching layer. I'm fine either way but we probably don't need this in the webhook validation.